astropy/coordinates/sky_coordinate.py

```diff
@@ -13,6 +13,7 @@
 from .. import units as u
 from ..wcs.utils import skycoord_to_pixel, pixel_to_skycoord
 from ..utils.exceptions import AstropyDeprecationWarning
+from ..utils.data_info import DataInfo, BaseInfo
```

Comment:
At some point I had a mental note to use `<classname>Info` but forgot to implement.  I was also struggling with naming, but I think you've hit on a good scheme so I'll do that.